### PR TITLE
New Spawns: StrawWagen's Placements

### PR DESCRIPTION
**Pull request by TechnichalDoge**

### CODE REVIEW

**Comment by TechnichalDoge** [comment]
[https://cdn.discordapp.com/attachments/347623692823560193/729839544324718672/20200706162243_1.jpg](url)

**Comment by TechnichalDoge** [comment]
[https://cdn.discordapp.com/attachments/574486617155436544/729846090542874665/20200706164850_1.jpg](url)

**Comment by overlordbleck** [comment]
What exactly is different here?

**Comment by TechnichalDoge** [comment]
New spawnpoints

**Comment by TechnichalDoge** [comment]
image

**Comment by TechnichalDoge** [comment]
courtesy of strawwagen

**Comment by TechnichalDoge** [comment]
at_the_20_hour

**Comment by StrawWagen** [comment]
replying to @overlordbleck , I redid the spawns for the map, (roughly) basing their placement off of the spawns on bigcity.

I transferred the outline of bigcity's spawn (represented with ropes in the screenshot logica posted) to a dupe and tried to fit it in an enclosed part of novenka, I decided that in the buildings would be good, and placed a continuous ring of spawns that outline an area very roughly equal to the area of bigcity's spawn.

I placed each spawnpoint next to a wall, so anyone who spawns at them will have a flank always covered.

I decided to use the footprint of bigcity's spawn since that map has really quick engagement times, and I feel that the current spawnpoints are too far apart.

**Comment by TechnichalDoge** [comment]
this seems to be the more discussion orientated one, so lets discuss this PR https://github.com/CFC-Servers/cfc_random_spawn/pull/13 with this discussion

**Comment by TechnichalDoge** [comment]
I chose the spawnpoints i did simularly, but not really to Straw. I tried to pick ones that would best fit if it was like bigciy, which has a big open space encircled by spawns and alleys. Instead of using a bigcity outline, i just expanded it to what seemed fitting to novenka

**Comment by TechnichalDoge** [comment]
i also think that the debate between these two should be voted on in announcements or something

**Comment by StrawWagen** [comment]
Here's the spawn locations from different angles, all the yellow poles are them.

**Comment by TechnichalDoge** [comment]
@StrawWagen could you post images of my version, as well?

**Comment by StrawWagen** [comment]
heeere they are @TechnichalDoge

**Comment by ghost** [comment]
Is this supposed to be merged with https://github.com/CFC-Servers/cfc_random_spawn/pull/13 or is this a choice between the two?

**Comment by StrawWagen** [comment]
the latest update has changes suggested by decline, as well it includes angles and pvp states of the spawns

**Comment by TechnichalDoge** [comment]
@Periapsises #13 was an alternative, the ones i suggested, so a choice, but people overwelmingly voted it down when posted with straw's in sugdisc

File: lua/cfc_random_spawn/sv_config.lua [hunk 1]
```diff
@@ -22,45 +22,68 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
...
-    { spawnPos = Vector( -2446, 7283, -384 ) }, -- Angle( 0, 45, 0 )
-    { spawnPos = Vector( -2580, 7039, -128 ) }, -- Angle( 0, -45, 0 )
-    { spawnPos = Vector( -2327, 6835, -128 ) }, -- Angle( 0, 135, 0 )
-    { spawnPos = Vector( -2636, 6797, -128 ) }, -- Angle( 0, 45, 0 )
-    { spawnPos = Vector( -2340, 8198, -384 ) }, -- Angle( 0, 0, 0 )
-    { spawnPos = Vector( -2435, 9145, -384 ) }, -- Angle( 0, 45, 0 )
-    { spawnPos = Vector( 3910, 7229, -184 ) }, -- Angle( 0, 135, 0 )
-    { spawnPos = Vector( 3872, 6256, -240 ) }, -- Angle( 0, 90, 0 )
-    { spawnPos = Vector( 2990, 8304, -264 ) } -- Angle( 0, 135, 0 )
+-- CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_novenka"] = { --original
```
Comment (plally) [inline lua/cfc_random_spawn/sv_config.lua]: we can always look through the github history if we need the old spawnpoints, commenting them probably isnt necessary

**Review by plally — CHANGES_REQUESTED** [review]

File: lua/cfc_random_spawn/sv_config.lua (hunk 1 above)
Comment (TechnichalDoge) [reply lua/cfc_random_spawn/sv_config.lua]: fair enough

**Review by plally — COMMENTED** [review]
this is fine , although im not sure the keys spawnAngle and pvp actually do anything. perhaps someone could provide more info as to why they were added?

it looks like the option to have pvp only spawns was removed a few commits back?

there could be value in keeping the pvp key if pvp only spawns are to be added back in the future

**Comment by StrawWagen** [comment]
@PLally phatso gave me a table to fill in, i filled it in.
https://discordapp.com/channels/225975453138026497/390548646233636865/730509597969743872

I wondered the same thing, but phatso's the person who has the answer to your question, it's probably for future-proofing

**Comment by sarahsturgeon** [comment]
We eventually want to also include the spawn angle. should be simple enough, just wanted to support it before the code was written

**Comment by Deconkle** [comment]
this code does not error

**Comment by StrawWagen** [comment]
looks like it's ready to merge then

**Comment by TechnichalDoge** [comment]
@Deconkle

**Comment by TechnichalDoge** [comment]
@overlordbleck @brandonsturgeon whoever gets here first can get the honor of merging! :smile:

File: lua/cfc_random_spawn/sv_config.lua [hunk 2]
```diff
@@ -22,42 +22,25 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
...
-    { spawnPos = Vector( -499, 7236, -384 ) }, -- Angle( 0, 90, 0 )
-    { spawnPos = Vector( -1042, 7049, -384 ) }, -- Angle( 0, 135, 0 )
-    { spawnPos = Vector( -2026, 7021, -384 ) }, -- Angle( 0, 45, 0 )
-    { spawnPos = Vector( -2446, 7283, -384 ) }, -- Angle( 0, 45, 0 )
-    { spawnPos = Vector( -2340, 8198, -384 ) }, -- Angle( 0, 0, 0 )
-    { spawnPos = Vector( -2435, 9145, -384 ) }, -- Angle( 0, 45, 0 )
-    { spawnPos = Vector( 3910, 7229, -184 ) }, -- Angle( 0, 135, 0 )
-    { spawnPos = Vector( 3872, 6256, -240 ) }, -- Angle( 0, 90, 0 )
-    { spawnPos = Vector( 2990, 8304, -264 ) } -- Angle( 0, 135, 0 )
+CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_novenka"] = { -- strawwagm_venka
```
Comment (overlordbleck) [inline lua/cfc_random_spawn/sv_config.lua]:
```suggestion
CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_novenka"] = {
```

**Review by overlordbleck — CHANGES_REQUESTED** [review]

File: lua/cfc_random_spawn/sv_config.lua [hunk 3]
```diff
@@ -22,42 +22,25 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
...
+    { spawnPos = Vector( 6052, 11677, -383 ), spawnAngle = Angle( 0, 0, -135 ), pvp = true },
```
Comment (overlordbleck) [inline lua/cfc_random_spawn/sv_config.lua]: Shouldn't the spawn angle be in the yaw and not the roll...? Unless we want people to spawn sideways I guess...

**Review by overlordbleck — CHANGES_REQUESTED** [review]

File: lua/cfc_random_spawn/sv_config.lua (hunk 3 above)
Comment (TechnichalDoge) [reply lua/cfc_random_spawn/sv_config.lua]: @brandonsturgeon @StrawWagen

File: lua/cfc_random_spawn/sv_config.lua [hunk 4]
```diff
@@ -23,41 +23,24 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
...
+    { spawnPos = Vector( 9043, 9799, -383 ), spawnAngle = Angle( 0, 124, 0 ), pvp = true },
+    { spawnPos = Vector( 7648, 10733, -383 ), spawnAngle = Angle( 0, -90, 0 ), pvp = true },
+    { spawnPos = Vector( 8486, 5880, -383 ), spawnAngle = Angle( 0, 155, 0 ), pvp = true },
+    { spawnPos = Vector( 7265, 5878, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },
+    { spawnPos = Vector( 5578, 6911, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },
+    { spawnPos = Vector( 5121, 8820, -127 ), spawnAngle = Angle( 0, 90, 0 ), pvp = true },
+    { spawnPos = Vector( 4475, 8824, -127 ), spawnAngle = Angle( 0, 59, 0 ), pvp = true },
+    { spawnPos = Vector( 4497, 9553, -127 ), spawnAngle = Angle( 0, 0, 8 ), pvp = true },
+    { spawnPos = Vector( 5654, 8670, -383 ), spawnAngle = Angle( 0, 0, 0 ), pvp = true },
+    { spawnPos = Vector( 5694, 7802, -383 ), spawnAngle = Angle( 0, 0, 0 ), pvp = true }
```
Comment (StrawWagen) [inline lua/cfc_random_spawn/sv_config.lua]:
```suggestion
    { spawnPos = Vector( 8989, 9103, -383 ), spawnAngle = Angle( 0, -135, 0 ), pvp = true },	
    { spawnPos = Vector( 9043, 9799, -383 ), spawnAngle = Angle( 0, 124, 0 ), pvp = true },	
    { spawnPos = Vector( 8984, 7792, -383 ), spawnAngle = Angle( 0, -180, 0 ), pvp = true },	
    { spawnPos = Vector( 9024, 10741, -383 ), spawnAngle = Angle( 0, -148, 0 ), pvp = true },	
    { spawnPos = Vector( 7648, 10733, -383 ), spawnAngle = Angle( 0, -90, 0 ), pvp = true },	
    { spawnPos = Vector( 8995, 6627, -383 ), spawnAngle = Angle( 0, 135, 0 ), pvp = true },	
    { spawnPos = Vector( 6364, 10733, -383 ), spawnAngle = Angle( 0, -90, 0 ), pvp = true },	
    { spawnPos = Vector( 8486, 5880, -383 ), spawnAngle = Angle( 0, 155, 0 ), pvp = true },	
    { spawnPos = Vector( 5654, 8670, -383 ), spawnAngle = Angle( 0, 0, 0 ), pvp = true },	
    { spawnPos = Vector( 5694, 7802, -383 ), spawnAngle = Angle( 0, 0, 0 ), pvp = true },	
    { spawnPos = Vector( 7265, 5878, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },	
    { spawnPos = Vector( 6052, 11677, -383 ), spawnAngle = Angle( 0, -135, 0 ), pvp = true },	
    { spawnPos = Vector( 5121, 8820, -127 ), spawnAngle = Angle( 0, 90, 0 ), pvp = true },	
    { spawnPos = Vector( 5578, 6911, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },	
    { spawnPos = Vector( 6417, 5878, -383 ), spawnAngle = Angle( 0, 135, 0 ), pvp = true },	
    { spawnPos = Vector( 4875, 10749, -383 ), spawnAngle = Angle( 0, 42, 0 ), pvp = true },	
    { spawnPos = Vector( 5334, 11687, -383 ), spawnAngle = Angle( 0, -45, 0 ), pvp = true },	
    { spawnPos = Vector( 4497, 9553, -127 ), spawnAngle = Angle( 0, 8, 0 ), pvp = true },	
    { spawnPos = Vector( 4475, 8824, -127 ), spawnAngle = Angle( 0, 59, 0 ), pvp = true },	
    { spawnPos = Vector( 4884, 11390, -383 ), spawnAngle = Angle( 0, -45, 0 ), pvp = true },	
```

Comment (TechnichalDoge) [reply lua/cfc_random_spawn/sv_config.lua]: wait, wtf is this do

Comment (StrawWagen) [reply lua/cfc_random_spawn/sv_config.lua]: i did the angle fix through code

File: lua/cfc_random_spawn/sv_config.lua [hunk 5]
```diff
@@ -23,41 +23,24 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
...
+    { spawnPos = Vector( 8984, 7792, -383 ), spawnAngle = Angle( 0, -180, 0 ), pvp = true },
+    { spawnPos = Vector( 8989, 9103, -383 ), spawnAngle = Angle( 0, -135, 0 ), pvp = true },
+    { spawnPos = Vector( 9043, 9799, -383 ), spawnAngle = Angle( 0, 124, 0 ), pvp = true },
+    { spawnPos = Vector( 7648, 10733, -383 ), spawnAngle = Angle( 0, -90, 0 ), pvp = true },
+    { spawnPos = Vector( 8486, 5880, -383 ), spawnAngle = Angle( 0, 155, 0 ), pvp = true },
+    { spawnPos = Vector( 7265, 5878, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },
+    { spawnPos = Vector( 5578, 6911, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },
+    { spawnPos = Vector( 5121, 8820, -127 ), spawnAngle = Angle( 0, 90, 0 ), pvp = true },
+    { spawnPos = Vector( 4475, 8824, -127 ), spawnAngle = Angle( 0, 59, 0 ), pvp = true },
+    { spawnPos = Vector( 4497, 9553, -127 ), spawnAngle = Angle( 0, 0, 8 ), pvp = true },
```
Comment (StrawWagen) [inline lua/cfc_random_spawn/sv_config.lua]:
```suggestion
    { spawnPos = Vector( 4497, 9553, -127 ), spawnAngle = Angle( 0, 8, 0 ), pvp = true },
```
Missed angle fix

**Review by sarahsturgeon — APPROVED** [review]
shore, why not

**Comment by sarahsturgeon** [comment]
ok to test

**Comment by TechnichalDoge** [comment]
"shore, why not"
if your drunk log out of github

**Comment by StrawWagen** [comment]
??????? why man

**Pull request merged**